easybuild/easyconfigs/p/pip/pip-20.0.2.eb

```diff
@@ -0,0 +1,39 @@
+easyblock = 'PythonBundle'
+
+name = 'pip'
+version = '20.0.2'
+
+homepage = 'https://pip.pypa.io'
+description = "pip is the package installer for Python."
+
+toolchain = SYSTEM
+
+allow_system_deps = [('Python', SYS_PYTHON_VERSION)]
+
+use_pip = False
+
+exts_default_options = {'source_urls': [PYPI_SOURCE]}
+
+exts_list = [
+    ('setuptools', '44.0.0', {
+        'source_tmpl': 'setuptools-%(version)s.zip',
+        'checksums': ['e5baf7723e5bb8382fc146e33032b241efc63314211a3a120aaa55d62d2bb008'],
+    }),
+    # stick to latest 0.33.x, 0.34.x does not work (UnicodeEncodeError, see https://github.com/pypa/wheel/issues/331)
+    ('wheel', '0.33.6', {
```

Comment:
This is supposedly fixed in 0.34.2, why not use that?

Reply:
I tried using `0.34.2`, didn't work for me...

Reply:
To not regret this later can we get a couple test reports with 34.2 first? Multiple people reported success with that so it might also be helpful for upstream and we have an easy way to get reproducible results from a couple environments.

Reply:
I investigated a bit and found this to be solely a Python2 issue (I guess we'll see more of them), see https://github.com/pypa/wheel/issues/331. A possible solution is to patch setuptools: https://github.com/pypa/setuptools/issues/2003

Reply:
I don't think it's worth the effort to patch `setuptools` for this.

Also, I don't expect `setuptools` to care much about Python 2 these days...

Reply:
Well they introduced a deprecation notice for 45.x with a promise to not drop it before April(?), so I'd wait a couple days for a response on the issue(s)

Reply:
I discussed this a bit with the guys from setuptools and they were pretty clear that `setup.py install` is no longer supported/deprecated and `pip install` shall be used. Also this method of installing pip is not recommended and `get-pip.py` shall be used instead. And indeed I was able to install all this with:

```
python get-pip.py --no-setuptools --no-wheel --ignore-installed --prefix /tmp/pipeb
PATH=/tmp/pipeb/bin PYTHONPATH=/tmp/pipeb/lib/python2.7/site-packages:$PYTHONPATH pip install --prefix=/tmp/pipeb setuptools==44.0.0
PATH=/tmp/pipeb/bin PYTHONPATH=/tmp/pipeb/lib/python2.7/site-packages:$PYTHONPATH pip install --prefix=/tmp/pipeb .
```

Hence my suggestion:
- use pip to install these packages
- when no pip is found (or always?), install the one from get-pip.py in the preinstallopts (e.g.), which will then be replaced by the one installed in the last step (pip installing pip)

Reply:
@boegel @Flamefire would be good to find a resolution for this one.